conda-store-server/conda_store_server/_internal/settings.py

```diff
@@ -0,0 +1,170 @@
+# Copyright (c) conda-store development team. All rights reserved.
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file.
+
+import functools
+from typing import Any, Callable, Dict
+
+import pydantic
+from sqlalchemy.orm import Session
+
+from conda_store_server import api
+from conda_store_server._internal import schema
+
+
+class Settings:
+    def __init__(self, db: Session, deployment_default: schema.Settings):
+        self.db = db
+        self.deployment_default = deployment_default.model_dump()
+
+    def _ensure_closed_session(func: Callable):
+        @functools.wraps(func)
+        def wrapper(self, *args, **kwargs):
+            result = func(self, *args, **kwargs)
+            self.db.close()
+            return result
+
+        return wrapper
+
+    @_ensure_closed_session
+    def set_settings(
+        self,
+        namespace: str = None,
+        environment_name: str = None,
+        data: Dict[str, Any] = {},
```

Comment:
```suggestion
        data: Dict[str, Any] | None = None,
```

I know it's not getting modified below, but it's still best to avoid mutable default arguments (it'll generate a lint error anyway if we ever turn on more rules). We need an `if data is None: ...` block below if this suggestion is made.